gigantum/gigantum-client
# 1646 share widget

## Pull Request for Issue #1646 

**Describe changes introduced by this PR (Bugs fixed, features added, docs updated, etc.)**:
This adds a feature for sharing a link that will import a users project and build it. 
Options for being dumped into a jupyterLab file have been added.

**Please check if the PR fulfills these requirements**

  - PR is set to merge to integration
  - PR title is descriptive, as this becomes the commit message on squash and merge.
  - Any related issues are added to the extended commit message during squash and merge.
  - Tests have been added (for bug fixes/features)
  - Docs have been added/updated (for bug fixes/features)

**QA Test Steps and Exit Criteria**:
1. Open a project and go to the actions menu (three ellipsis on the top right)
2. Click on share link to pop the modal
3. Under tools select jupyterLab to enable file sharing, property should update in text area above 
4. Check that auto complete works (type in `ipynb` if that file type is present) on code
5. Make sure that autocomplete adds file to the text box when clicked on and when using a mouse.
6. copy link using the copy button, paste link into a new tab and confirm it works as expected.
